cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -395,6 +433,10 @@ void AddMinMaxKernels(KernelInit init,
     auto out_ty = struct_({field("min", ty), field("max", ty)});
     auto sig = KernelSignature::Make({InputType::Array(ty)}, ValueDescr::Scalar(out_ty));
     AddAggKernel(std::move(sig), init, func, simd_level);
+
+    // scalar[InT] -> scalar[struct<min: T, max: T>]
+    sig = KernelSignature::Make({InputType::Scalar(ty)}, ValueDescr::Scalar(out_ty));
+    AddAggKernel(std::move(sig), init, func, SimdLevel::NONE);
```

Comment:
Instead since the same init is being used for each kernel, I'd prefer we just make the signature inclusive of scalars as well as arrays:
```suggestion
    auto sig = KernelSignature::Make({InputType(ty)}, ValueDescr::Scalar(out_ty));
    AddAggKernel(std::move(sig), init, func, simd_level);
```